src/background.js

```diff
@@ -1,9 +1,10 @@
 "use strict";
 /* global browser, runHeuristics */
 
-function log() {
+async function log() {
   // eslint-disable-next-line no-constant-condition
-  if (false) {
+  const showConsoleLogs = await browser.experiments.preferences.getBoolPref("doh-rollout.debug", false);
```

Comment:
This is a bit of a perf hit on the addon for logging, can you check it once and make it a global instead?

Reply:
If you store a pref value outside of the pref system, make sure to keep it updated with a pref watcher.